frame/contracts/src/wasm/code_cache.rs

```diff
@@ -118,6 +118,16 @@ pub fn decrement_refcount<T: Config>(code_hash: CodeHash<T>) -> Result<(), Dispa
 	Ok(())
 }
 
+/// Increment the refcount of a code in-storage by one.
+pub fn increment_refcount<T: Config>(code_hash: CodeHash<T>) -> Result<(), DispatchError> {
+	<OwnerInfoOf<T>>::mutate(code_hash, |existing| {
+		if let Some(info) = existing {
+			info.refcount = info.refcount.saturating_add(1);
+		}
+	});
+	Ok(())
+}
```

Comment:
Make it return an error if the entry doesn't exist ( `existing = None`). Then we can use that as an error condition when setting the new code. Please make sure to document that behavior.

The return type and comment on `decrement_refcount` is wrong by the way. I forgot to update it. It is infallible and doesn't delete anything when it drops to zero. Just so you don't get confused.

Reply:
Making update

Won't touch `decrement_refcount` right now, as it is used elsewhere as well